### PR TITLE
sporemind

### PR DESCRIPTION
**Pull request by MadmanMartian**
adds the sporemind, which lets mushroms talk to one another like ancientchat or cult/ling chat. Use :~ to talk with it as a mushrom or a minimush.

sporetalk

:cl:
* rscadd: Adds the sporemind global chat for mushroom people. Use .~ to talk with it

### CODE REVIEW

File: __DEFINES/_macros.dm [hunk 1]
```diff
@@ -38,6 +38,8 @@
 
 #define isgrue(A) (ishuman(A) && A.species && istype(A.species, /datum/species/grue))
 
+#define ismushroom(A) ((ishuman(A) && A.species && istype(A.species, /datum/species/mushroom)) || (ismonkey(A) && istype(A, /mob/living/carbon/monkey/mushroom)))
```
Comment (DamianX) [inline __DEFINES/_macros.dm]: Why bother checking for `ismonkey` if you're already checking for `/mob/living/carbon/monkey/mushroom`?

**Comment by Pathid** [comment]
This solves not being able to talk to the baby mushrooms. Nice.

**Comment by DamianX** [comment]
pls fix?

**Comment by MadmanMartian** [comment]
fixd

**Pull request merged**